### DISCUSSION TITLE
Total number of pages (Pages length)

### DISCUSSION THREAD

**Discussion by 411A** [typst/typst#4084] [Q&A] [answered]
Hi, I want to retrieve the total number of pages. Unfortunately, I couldn’t find this information in the documentation, or perhaps I don’t know how to access the desired value. My goal is to create a custom page numbering display inside footer using the following format (which only shows current page number):
```typ
#counter(page).display() / #counter(page).last(loc)
```
Desired output:
_page_number / total_pages_
```
1 / 3
```
But the `#counter(page).last(loc)` is not working (results in no output, **type counter has no method `last`**)...

**Comment by KillTheMule** [comment]
Try 

```typst
#counter(page).display("1") / #locate(loc => {
          let lastpage = counter(page).final(loc)
          [#lastpage.at(0)]
        })
```

**Reply by 411A** [reply]
Thanks! It works on an empty page, but I used [this template](https://github.com/bamboovir/typst-resume-template/blob/main/template.typ#L67), it doesn't work here...

**Reply by KillTheMule** [reply]
Works in my document :) Sorry, no idea what the interaction here could be.

**Comment by flokl** [comment, marked as answer]
Hello, the following should work.
You can find more information at: https://typst.app/docs/guides/page-setup-guide/#custom-footer-with-page-numbers
and https://typst.app/docs/reference/introspection/counter/#definitions-display.

```
#context counter(page).display(
  "1/1",
  both: true,
)
```

**Reply by 411A** [reply]
Thank you so much!
